test/Nerdbank.MessagePack.Analyzers.Tests/MigrationAnalyzerTests.cs

```diff
@@ -0,0 +1,147 @@
+			{
+				public MyType Deserialize(ref MessagePackReader reader, MessagePackSerializerOptions options)
+				{
+					if (reader.TryReadNil())
+					{
+						return null;
+					}
+
+					string name = null;
+					options.Security.DepthStep(ref reader);
+					try
+					{
+						int count = reader.ReadArrayHeader();
+						for (int i = 0; i < count; i++)
+						{
+							switch (i)
+							{
+								case 0:
+									name = options.Resolver.GetFormatterWithVerify<string>().Deserialize(ref reader, options);
+									break;
+								default:
+									reader.Skip();
+									break;
+								}
+						}
+
+						return new MyType { Name = name };
+					}
+					finally
+					{
+						reader.Depth--;
+					}
+				}
+
+				public void Serialize(ref MessagePackWriter writer, MyType value, MessagePackSerializerOptions options)
+				{
+					if (value is null)
+					{
+						writer.WriteNil();
+						return;
+					}
+
+					writer.WriteArrayHeader(1);
+					options.Resolver.GetFormatterWithVerify<string>().Serialize(ref writer, value.Name, options);
+				}
+			}
+			""";
+
+		string fixedSource = /* lang=c#-test */ """
+			using MessagePack;
+			using MessagePack.Formatters;
+			using Nerdbank.MessagePack;
+			using PolyType;
+			
+			[MessagePackConverter(typeof(MyTypeFormatter))]
+			public class MyType
+			{
+				public string Name { get; set; }
+			}
+
+			[GenerateShape<string>]
+			public partial class MyTypeFormatter : MessagePackConverter<MyType>
+			{
+				public override MyType Deserialize(ref Nerdbank.MessagePack.MessagePackReader reader, SerializationContext context)
+				{
+					if (reader.TryReadNil())
+					{
+						return null;
+					}
+
+					string name = null;
+					context.DepthStep();
+					int count = reader.ReadArrayHeader();
+					for (int i = 0; i < count; i++)
+					{
+						switch (i)
+						{
+							case 0:
+								name = context.GetConverter<string, MyTypeFormatter>().Deserialize(ref reader, context);
+								break;
+							default:
+								reader.Skip(context);
+								break;
+						}
+					}
+
+					return new MyType { Name = name };
+				}
+
+				public override void Serialize(ref Nerdbank.MessagePack.MessagePackWriter writer, in MyType value, SerializationContext context)
+				{
+					if (value is null)
+					{
+						writer.WriteNil();
+						return;
+					}
+
+					writer.WriteArrayHeader(1);
+					context.GetConverter<string, MyTypeFormatter>().Serialize(ref writer, value.Name, context);
+				}
+			}
+			""";
+
+		await this.VerifyCodeFixAsync(source, fixedSource);
+	}
+
+	[Fact(Skip = "Not implemented")]
+	public void FormatterWithExtensionReferences()
+	{
+		// Verify migration when the formatter accesses types and members with the old library's
+		// Extension*Format* naming pattern and update it to drop the 'Format' portion.
+	}
+
+	private Task VerifyCodeFixAsync([StringSyntax("c#-test")] string source, [StringSyntax("c#-test")] string fixedSource)
+	{
+		return new VerifyCS.Test
```

Comment:
@sharwell How do I inject an external source generator into the test harness? The "fixed" code only compiles when combined with a source generator. Right now, that's the only thing keeping my test from passing.